lib/hashes.md

```diff
@@ -5,40 +5,42 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
-Replace me with your response...
+Since hashes do not store the items in a numbered list, so if you need the data stored in a specific order, so that it can be called by referencing a specific number in that sequence, then hashes can be useful.  They can group and organize data/objects,
+and make it so that items can be called without havinig to memorize their specific number in the ordered list or array.  Arrays store named data, so the use of the name can do a lot of things with the data associated with that key.  A type of data that can 
+benefit from the use of hashes is counting data, as it can store amounts of a certain type of item (as opposed to arrays).
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
-
+Some of this response could be found in the previous answer above, but I'll try to repeat as much here.  Both arrays and hashes are collections of information, stored in a form of repository.  The differences come mainly from how that data is stored, and how that data can be used or called.
+Arrays store the data/objects in a specific order, and particular items can be called by utilizing the number of the item in that list (beginning with zero).  Hashes have keys, which are unique names that can be used to identify values.  The keys can be used to help do a variery of things, like 
+organize a count of like items, or categorize items into desired groups, etc.  Hashes have much more variety in terms of what they can offer.
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
-
+A particular value from a hash is retrieved by using the key for that value.  For instance, if this is our hash:  grades = { "Mike Ducey" => A, "Pat Ducey" => B+ }, you reference the hash like an array, but instead of specifying the number of the item, you use the key.  So, in this case, if I write grades["Mike Ducey"], I shall get => A as a response.
 # How do you add information to a Hash?
 
-Replace me with your response...
-
+This is done by using the "bracket-equals method" (https://github.com/learn-co-curriculum/hash-overview-readme).  That is, if you have a hash and you want another key in the hash, you can write the name of the key in the brackets and the value after the equal sign.  I'll add a grade to my grades hash: grades["Kevin Ducey"] = A-
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+This is achieved by using the each function.  If you want to change each value, you use something of the form: grades.each do |x| where x is the change to be done using code.
 
 # How would you change the value of a particular element in a Hash?
 
-Replace me with your response...
+This is done via utilizing the same sort of method as assigning a new key/value pair.  So, it is basically an overwriting of a value.  I could write grades("Pat Ducey") = C+ and that would be the new value for the key "Pat Ducey"
```

Comment:
Ack!  I meant to write grades.update("Pat Ducey" +> "C+")  That would update that key-value pair. 

I noticed I didn't get the form right on the values for those grades.  Without quotation marks, it definitely produces errors.